internal/cmd/shell/shell.go

```diff
@@ -272,58 +271,17 @@ func (m *mysql) Run(ctx context.Context, sigc chan os.Signal, signals []os.Signa
 	c.Stderr = os.Stderr
 	c.Stdin = os.Stdin
 
-	// Set up a new channel for signals received while MySQL is active.
-	// This is registered for all signals so we forward them all to MySQL,
-	// so we behave as much as possible like a regular MySQL.
-	// When we exit this function, we stop the custom signal receiver.
-	msig := make(chan os.Signal, 1)
-	signal.Notify(msig)
-	defer signal.Stop(msig)
-
-	// We stop handling signals for our default setup from the CLI. This
-	// is needed, so we stop handling for example the default os.Interrupt
-	// that stops the shell and we forward it to MySQL.
-	// When we exit from this function, we restore the signals as they were.
-	signal.Stop(sigc)
-	defer signal.Notify(sigc, signals...)
-
-	err := c.Start()
-	if err != nil {
-		return err
-	}
+	c.SysProcAttr = sysProcAttr()
+	setupSignals(ctx, c, sigc, signals)
```

Comment:
Why is the return value not used? I believe it's non-nil for Windows.

Reply:
Good catch, let me fix this up.